contentcuration/contentcuration/tasks.py

```diff
@@ -165,3 +177,219 @@ def sendcustomemails_task(subject, message, query):
+        )
+    except License.DoesNotExist:
+        logger.warning("License 'Special Permissions' not found in database")
+        return []
+    except License.MultipleObjectsReturned:
+        logger.warning("Multiple 'Special Permissions' licenses found, using first one")
+        special_permissions_license = License.objects.filter(
+            license_name="Special Permissions"
+        ).first()
+
+    if (
+        not special_permissions_license
+        or special_permissions_license.id not in included_licenses
+    ):
+        return []
+
+    unversioned_db_filename = f"{channel_id}.sqlite3"
+    unversioned_db_storage_path = os.path.join(
+        settings.DB_ROOT, unversioned_db_filename
+    )
+
+    if not storage.exists(unversioned_db_storage_path):
+        logger.error(
+            f"Unversioned database not found at {unversioned_db_storage_path} for channel {channel_id}"
+        )
+        return None
+
+    with using_temp_migrated_content_database(unversioned_db_storage_path):
+        special_perms_nodes = KolibriContentNode.objects.filter(
+            license_name="Special Permissions"
+        ).exclude(kind=content_kinds.TOPIC)
+
+        license_descriptions = (
+            special_perms_nodes.exclude(license_description__isnull=True)
+            .exclude(license_description="")
+            .values_list("license_description", flat=True)
+            .distinct()
+        )
+
+        audited_license_ids = []
+        for description in license_descriptions:
+            (
+                audited_license,
+                created,
+            ) = AuditedSpecialPermissionsLicense.objects.get_or_create(
+                description=description, defaults={"distributable": False}
+            )
+            audited_license_ids.append(audited_license.id)
+            if created:
+                logger.info(
+                    f"Created new AuditedSpecialPermissionsLicense for description: {description[:100]}"
+                )
+
+        return audited_license_ids
+
+
+def _save_audit_results(
+    channel,
+    published_data_version,
+    invalid_license_ids,
+    special_permissions_license_ids,
+    user_id,
+):
+    """Save audit results to published_data and create change event."""
+    published_data_version["community_library_invalid_licenses"] = (
+        invalid_license_ids if invalid_license_ids else None
+    )
+    published_data_version["community_library_special_permissions"] = (
+        special_permissions_license_ids if special_permissions_license_ids else None
+    )
+
+    channel.save()
+
+    Change.create_change(
+        generate_update_event(
+            channel.id,
+            CHANNEL,
+            {"published_data": channel.published_data},
+            channel_id=channel.id,
+        ),
+        applied=True,
+        created_by_id=user_id,
+    )
+
+
+@app.task(bind=True, name="audit-channel-licenses")
+def audit_channel_licenses_task(self, channel_id, user_id):
```

Comment:
Is there any benefit in binding the function to the celery task model? It seems we are not using the `self` argument, so we can perhaps just skip it.